doc-source/quick-start.rst

```diff
@@ -64,6 +64,43 @@ As an example, the format for ``controls.json`` is as follows::
     "chk_pkg.chk_cat_bar.checks.chk_module_bar.BarCheckClass": ["accred.one", "accred.two"]
   }
 
+Agents
+~~~~~~
+
+All fetchers can be executed in "agent" mode. Agents will cryptographically sign
+any evidence they fetch. The agent name, evidence digest and signature can be
+found in the ``index.json`` metadata file.
+
+Signed evidence can be used in checks. It is automatically verified when it's
+loaded from the locker. However, the public key used to verify the evidence must
+be made available in the locker under ``raw/auditree/agent_public_keys.json``.
+This evidence takes the following form::
+
+  {
+    "my-agent-name": "-----BEGIN PUBLIC KEY-----\n...",
+    "my-other-agent-name": "-----BEGIN PUBLIC KEY-----\n...",
+    ...
+  }
+
+We recommend you implement an additional fetcher to pull this public key
+evidence into the locker and ensure it's kept up-to-date.
+
+To configure an agent, add the following to your main configuration::
+
+  {
+    "agent_name": "my-agent-name",
+    "agent_private_key": "/path/to/key.pem",
+    ...
+  }
+
+Each agent must have a unique name. By default, any evidence created by an agent
+will be stored under the corresponding agent directory (e.g.
+``agents/my-agent-name/raw/system/uptime.txt``). Set ``"use_agent_dir": false``
+to prevent this.
+
+Only use agents for executing fetchers. Running checks when configured as an
+agent is not supported.
```

Comment:
As discussed elsewhere, I can imagine a future whether the CI job signs it's evidence as well. In that case, an agent would be running both fetchers and checks.

Reply:
Yep, I'll clarify that.